hive/build.gradle

```diff
@@ -21,6 +21,14 @@ jar {
     }
 }
 
+itestJar {
+    from(zipTree(project(":elasticsearch-hadoop-mr").jar.archivePath)) {
```

Comment:
I think the preferred way to model this is to via configuration/artifacts/dependencies instead of referencing the project directly. 

So ... this build.gradle would looks similar to
```
configurations {
    sharedHadoop
    sharedSpi
}
...
itestJar {
    from configurations.sharedHadoop
    from configurations.sharedSpi
}
...
dependencies {
 ...
    sharedHadoop project(path: ':elasticsearch-hadoop-mr', configuration: 'sharedHadoop')
    sharedSpi project(path: ':elasticsearch-hadoop-mr', configuration: 'sharedSpi')
}
```
and mr/build.gradle would look something like
```
configurations {
    sharedHadoop
    sharedSpi
}

artifacts{
    sharedHadoop(new File(projectDir, "src/main/java/org/elasticsearch/hadoop"))
    sharedSpi(new File(projectDir, "src/main/resources/META-INF/services"))
}
```

(and don't have an answer for the generated esh-build.properties)

However, implementing something like that is probably outside the scope of this PR especially since what you have here matches the existing model.

Reply:
This work is already in progress, spread across multiple changes at the moment. This work sort of highlighted the problem. I can combine those PR's together but I would like to keep them split up so they're easier to review.